can not build github repo with cmake on MacOS 15.5

seems there are two minor places need to be fixed.

steps to reproduce:

```
git clone https://github.com/libarchive/libarchive.git
cd libarchive
cmake .
make -j
```

output:

<img width="650" height="226" alt="Image" src="https://github.com/user-attachments/assets/3d234e1c-04f0-45c6-8976-74a1abe34a51" />

<img width="649" height="350" alt="Image" src="https://github.com/user-attachments/assets/0231d377-27c4-4f49-9c0e-63d44bd09084" />

I found that the errors are caused by gcc

I installed gcc by homebrew, and set the CC and CXX environment macro in `~/.zprofile`

```
export CC="$(brew --prefix gcc@15)/bin/gcc-15"
export CXX="$(brew --prefix gcc@15)/bin/g++-15"
```

then I get the compile command from `make VERBOSE=1` 
```
/opt/homebrew/opt/gcc/bin/gcc-15 -DHAVE_CONFIG_H -D__LIBARCHIVE_ENABLE_VISIBILITY -Darchive_EXPORTS -I/Users/cjn/src/libarchive/libarchive -I/Users/cjn/src/libarchive -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -I/opt/homebrew/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libxml2 -I/Users/cjn/src/libarchive/libarchive/. -Wall -Wformat -Wformat-security -Werror -fvisibility=hidden -g -g -Wextra -Wunused -Wshadow -Wmissing-prototypes -Wcast-qual -arch arm64 -fPIC -MD -MT libarchive/CMakeFiles/archive.dir/archive_disk_acl_darwin.c.o -MF CMakeFiles/archive.dir/archive_disk_acl_darwin.c.o.d -o CMakeFiles/archive.dir/archive_disk_acl_darwin.c.o -c /Users/cjn/src/libarchive/libarchive/archive_disk_acl_darwin.c
/Users/cjn/src/libarchive/libarchive/archive_disk_acl_darwin.c: In function 'add_trivial_nfs4_acl':
/Users/cjn/src/libarchive/libarchive/archive_disk_acl_darwin.c:198:23: error: comparison of integer expressions of different signedness: 'int' and 'long unsigned int' [-Werror=sign-compare]
  198 |         for (i = 0; i < sizeof(tacl_entry) / sizeof(tacl_entry[0]); i++) {
      |                       ^
/Users/cjn/src/libarchive/libarchive/archive_disk_acl_darwin.c: At top level:
/Users/cjn/src/libarchive/libarchive/archive_disk_acl_darwin.c:558:1: fatal error: opening dependency file CMakeFiles/archive.dir/archive_disk_acl_darwin.c.o.d: No such file or directory
  558 | }
      | ^
compilation terminated.
cc1: all warnings being treated as errors
```

it is OK if I commented the CC/CXX in .zprofile and  compile with clang. 

So there is another question: how to build with gcc on MacOC?

--- a/libarchive/archive_disk_acl_darwin.c
+++ b/libarchive/archive_disk_acl_darwin.c
@@ -124,7 +124,6 @@ static void
 add_trivial_nfs4_acl(struct archive_entry *entry)
 {
 	mode_t mode;
-	int i;
 	const int rperm = ARCHIVE_ENTRY_ACL_READ_DATA;
 	const int wperm = ARCHIVE_ENTRY_ACL_WRITE_DATA |
 	    ARCHIVE_ENTRY_ACL_APPEND_DATA;
@@ -195,7 +194,7 @@ add_trivial_nfs4_acl(struct archive_entry *entry)
 	} else if ((mode & 0010) || (mode & 0001))
 		tacl_entry[1].permset |= eperm;
 
-	for (i = 0; i < sizeof(tacl_entry) / sizeof(tacl_entry[0]); i++) {
+	for (size_t i = 0; i < sizeof(tacl_entry) / sizeof(tacl_entry[0]); i++) {
 		if (tacl_entry[i].permset != 0) {
 			archive_entry_acl_add_entry(entry,
 			    tacl_entry[i].type, tacl_entry[i].permset,

